Formula/c/codecov-cli.rb

```diff
@@ -0,0 +1,142 @@
+class CodecovCli < Formula
+  include Language::Python::Virtualenv
+
+  desc "Codecov's command-line interface"
+  homepage "https://cli.codecov.io/"
+  url "https://github.com/codecov/codecov-cli/archive/refs/tags/v0.6.0.tar.gz"
+  sha256 "90f54ea6dabb1682b3f6a84fa45cded79c585b25fee739908e26e16525937239"
+  license "Apache-2.0"
+  head "https://github.com/codecov/codecov-cli.git", branch: "main"
+
+  depends_on "rust" => :build
+  depends_on "python@3.11"
```

Comment:
I was referring to the build being used by `codecov/codecov-cli` in [my previous comment](https://github.com/Homebrew/homebrew-core/pull/174363#issuecomment-2163456769), not in `tree-sitter/py-tree-sitter`.

Reply:
Copying that comment here for brevity:
> The [upstream application is being built with Python 3.11](https://github.com/codecov/codecov-cli/blob/v0.6.0/.github/workflows/build_assets.yml#L56-L59) so I've made some changes to accommodate that. Should prevent some of the errors which have been thrown.

Reply:
Yup, but I'm referring to the issue referenced here: https://github.com/codecov/codecov-cli/blob/8c869787ab875e0f2a8468a7ad16124d0b6de5db/.github/workflows/ci.yml#L40

> Should prevent some of the errors which have been thrown

But what errors exactly? I'm asking because I've pulled this PR down locally and it builds + tests fine after my suggested changes

Reply:
Apologies, I should have included them in that message at the time. The `Regex` module was complaining about the format being used for some of the regular expressions, I assume due to changes between Python 3.11 and 3.12.

Reply:
I can reproduce this error after installing the formula with your suggested changes:
```
/opt/homebrew/opt/codecov-cli/bin/codecovcli --help
/opt/homebrew/Cellar/codecov-cli/0.6.0/libexec/lib/python3.12/site-packages/codecov_cli/helpers/ci_adapters/codebuild.py:15: SyntaxWarning: invalid escape sequence '\/'
  return re.sub("^refs\/heads\/", "", branch)
/opt/homebrew/Cellar/codecov-cli/0.6.0/libexec/lib/python3.12/site-packages/codecov_cli/helpers/ci_adapters/codebuild.py:30: SyntaxWarning: invalid escape sequence '\/'
  slug = re.sub("^.*github.com\/", "", slug)
/opt/homebrew/Cellar/codecov-cli/0.6.0/libexec/lib/python3.12/site-packages/codecov_cli/helpers/ci_adapters/codebuild.py:31: SyntaxWarning: invalid escape sequence '\/'
  slug = re.sub("^.*gitlab.com\/", "", slug)
/opt/homebrew/Cellar/codecov-cli/0.6.0/libexec/lib/python3.12/site-packages/codecov_cli/helpers/ci_adapters/codebuild.py:32: SyntaxWarning: invalid escape sequence '\/'
  slug = re.sub("^.*bitbucket.com\/", "", slug)
/opt/homebrew/Cellar/codecov-cli/0.6.0/libexec/lib/python3.12/site-packages/codecov_cli/helpers/ci_adapters/codebuild.py:33: SyntaxWarning: invalid escape sequence '\.'
  return re.sub("\.git$", "", slug)
/opt/homebrew/Cellar/codecov-cli/0.6.0/libexec/lib/python3.12/site-packages/codecov_cli/helpers/ci_adapters/codebuild.py:42: SyntaxWarning: invalid escape sequence '\/'
  return re.sub("^pr\/", "", pr)
Usage: codecovcli [OPTIONS] COMMAND [ARGS]...

Options:
  --auto-load-params-from [CircleCI|GithubActions|GitlabCI|Bitbucket|Bitrise|AppVeyor|Woodpecker|Heroku|DroneCI|BuildKite|AzurePipelines|Jenkins|CirrusCI|Teamcity|Travis|AWSCodeBuild|Local]
  --codecov-yml-path PATH
  -u, --enterprise-url, --url TEXT
                                  Change the upload host (Enterprise use)
  -v, --verbose                   Use verbose logging
  --version                       Show the version and exit.
  --help                          Show this message and exit.

Commands:
  create-commit
  create-report
  create-report-results
  do-upload
  empty-upload
  get-report-results
  label-analysis
  pr-base-picking
  process-test-results
  send-notifications
  static-analysis
  upload-process
```

Reply:
That being said, my Python 3.12 version is slightly out of date so let me update that and reinstall this formula to confirm that problem actually exists.

Reply:
I'm seeing the same problem after reinstalling this formula with Python 3.12.4.

Reply:
Ah thanks, I see that now too. Filed https://github.com/codecov/codecov-cli/pull/458 if you want to add the first commit as a patch (typically, we don't allow patches for a new formula so CI might complain but I'm personally fine with it here)

Reply:
Done.

Reply:
The build is failing due to the `brew audit` check. What do you suggest as a workaround? Should I keep this patch in?

Reply:
yes no worries, i've added a label to let ci continue